packages/odf/components/s3-browser/upload-objects/upload-component/uploads.ts

```diff
@@ -1,76 +1,71 @@
+};
+
 export const uploadFile = async (
   files: File[],
   client: S3Commands,
   bucketName: string,
-  setUploadProgress: React.Dispatch<React.SetStateAction<UploadProgress>>,
-  folderPath: string
+  folderPath: string,
+  uploadStore: UploadStore
 ) => {
-  const performUploadPromise = (
-    file: File
-  ): ReturnType<typeof Upload.prototype.done> => {
-    const key = getPrefix(file.webkitRelativePath || file.name, folderPath);
-    const uploader = client.getUploader(file, key, bucketName);
-    uploader.on('httpUploadProgress', (progress) => {
-      const isComplete = progress.total === progress.loaded;
-      setUploadProgress((prev) => ({
-        ...prev,
-        [progress.Key]: {
-          startTime: prev[progress.Key].startTime ?? Date.now(),
-          name: progress.Key,
-          total: progress.total,
-          loaded: progress.loaded,
-          abort: () => {
-            uploader.abort();
-            setUploadProgress((previous) => {
-              const clonedPrev = _.cloneDeep(previous);
-              clonedPrev[progress.Key].uploadState = UploadStatus.UPLOAD_FAILED;
-              return clonedPrev;
-            });
-          },
-          uploadState: isComplete
-            ? UploadStatus.UPLOAD_COMPLETE
-            : UploadStatus.UPLOAD_START,
-        },
-      }));
-    });
-    return uploader.done();
-  };
-  const allUploadPromise = files.map((file) => performUploadPromise(file));
+  const allUploadPromise = files
+    .filter((file) => {
+      const key = getPrefix(file.webkitRelativePath || file.name, folderPath);
+      const item = uploadStore.getFile(key);
+      return item.uploadState !== UploadStatus.UPLOAD_CANCELLED;
+    })
+    .map((file) =>
+      performUploadPromise(file, folderPath, bucketName, uploadStore, client)
+    );
   const settledPromises = await Promise.allSettled(allUploadPromise);
   settledPromises.forEach((promise, i) => {
     const hasFailed = promise.status === 'rejected';
-    if (hasFailed) {
-      const key = getPrefix(
-        files[i].webkitRelativePath || files[i].name,
-        folderPath
-      );
-      setUploadProgress((prev) => ({
-        ...prev,
-        [key]: {
-          ...prev[key],
-          uploadState: UploadStatus.UPLOAD_FAILED,
-        },
-      }));
+    const key = getPrefix(
+      files[i].webkitRelativePath || files[i].name,
+      folderPath
+    );
+    const item = uploadStore.getFile(key);
+    const status = item.uploadState;
+    if (hasFailed && status !== UploadStatus.UPLOAD_COMPLETE) {
+      uploadStore.uploadFailed = key;
     }
   });
   return Date.now();
 };
 
 export const convertFileToUploadProgress = (
-  file: File
-): UploadProgress[keyof UploadProgress] => ({
-  total: file.size,
+  file: File,
+  key: string
+): UploadProgress => ({
   loaded: 0,
   uploadState: UploadStatus.INIT_STATE,
   abort: null,
   name: file.name,
   filePath: file.webkitRelativePath,
   startTime: undefined,
+  webkitRelativePath: '',
```

Comment:
shouldn't this be the case ??
```suggestion
  webkitRelativePath: file.webkitRelativePath,
```

Reply:
also if this `webkitRelativePath` is unused key, can we plz remove this entirely ??